src/Middleware/HeaderPropagation/src/HeaderPropagationEntry.cs

```diff
@@ -0,0 +1,15 @@
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.HeaderPropagation
+{
+    public class HeaderPropagationEntry
+    {
+        public string InputName { get; set; }
+        public string OutputName { get; set; }
+        public StringValues DefaultValues { get; set; }
+        public Func<HttpContext, StringValues> DefaultValuesGenerator { get; set; }
+        public bool AlwaysAdd { get; set; }
```

Comment:
Do you have a use case in mind for `AlwaysAdd`?  Usually you have a global default (propagate this header) and allow local overrides (specifying the header explicitly). Why do you think this is needed?

Reply:
see https://github.com/aspnet/AspNetCore/pull/7921#issuecomment-470457767

Reply:
> I'm tempted to leave this out for now (as it can eventually be added later as a non breaking change) eventually adding some extensibility points to be able to inject a behavior for the specific scenario. What do you think?

Yeah, I'd suggest leaving this out. Users will ask how to do it if it's important.

Another suggestion that I have for this (and other needs) - replace `DefaultValuesGenerator` with `ValueFactory` - if it's set, then you just always call it to get the output value. I think that solves all of the cases you were interested in, and also can fill the niche that it currently does.